poc/tests/test_vdaf_ping_pong.py

```diff
@@ -236,43 +237,52 @@ def test_multi_round(self) -> None:
                 rand,
             )
 
-            (leader_state, msg) = vdaf.ping_pong_leader_init(
+            leader_state: State = vdaf.ping_pong_leader_init(
                 verify_key,
                 ctx,
                 vdaf.encode_agg_param(agg_param),
                 nonce,
                 vdaf.test_vec_encode_public_share(public_share),
                 vdaf.test_vec_encode_input_share(input_shares[0]),
             )
-            self.assertEqual(leader_state, Continued((0, measurement), 0))
+            assert isinstance(leader_state, Continued)
+            self.assertEqual(leader_state.prep_round, 0)
 
             for step in range(num_steps):
                 if step == 0:
-                    (helper_state, msg) = vdaf.ping_pong_helper_init(
+                    assert isinstance(leader_state, Continued)
+                    helper_state: State = vdaf.ping_pong_helper_init(
                         verify_key,
                         ctx,
                         vdaf.encode_agg_param(agg_param),
                         nonce,
                         vdaf.test_vec_encode_public_share(public_share),
                         vdaf.test_vec_encode_input_share(input_shares[1]),
-                        cast(bytes, msg),
+                        leader_state.outbound,
                     )
                 else:
-                    (helper_state, msg) = vdaf.ping_pong_helper_continued(
+                    assert isinstance(leader_state, Continued) or \
+                        isinstance(leader_state, FinishedWithOutbound)
+                    helper_state = vdaf.ping_pong_helper_continued(
                         vdaf.encode_agg_param(agg_param),
                         ctx,
-                        helper_state,
-                        cast(bytes, msg),
+                        cast(Continued, helper_state),
+                        leader_state.outbound,
                     )
 
                 if isinstance(leader_state, Continued):
-                    (leader_state, msg) = vdaf.ping_pong_leader_continued(
+                    assert isinstance(helper_state, Continued) or \
+                        isinstance(helper_state, FinishedWithOutbound)
+                    leader_state = vdaf.ping_pong_leader_continued(
                         vdaf.encode_agg_param(agg_param),
                         ctx,
-                        leader_state,
-                        cast(bytes, msg),
+                        cast(Continued, leader_state),
```

Comment:
This cast can be removed, because the enclosing `if` narrows the inferred type here.
```suggestion
                        leader_state,
```